app/src/main/java/org/p2p/wallet/receive/widget/QrView.kt

```diff
@@ -111,8 +113,15 @@ class QrView @JvmOverloads constructor(
         binding.root.startAnimation(animation)
     }
 
-    private fun onSnapshotReady(bitmap: Bitmap) {
-        val name = binding.nameTextView.text.toString()
-        onSaveClickListener?.invoke(name, bitmap)
+    private fun onSnapshotReady(bitmap: Bitmap, action: QRCodeAction) {
+        val qrValue = binding.valueTextView.text.toString()
+        when (action) {
+            QRCodeAction.SHARE -> onShareClickListener?.invoke(qrValue, bitmap)
+            QRCodeAction.SAVE -> onSaveClickListener?.invoke(qrValue, bitmap)
```

Comment:
We have to recycler bitmap after using it, do we have that? Couldn't find in changes

Reply:
will add in method where we save QrBitmapToFile